presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/DefaultThriftMetastoreClientFactory.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.metastore.thrift;
+
+import com.google.common.net.HostAndPort;
+import io.airlift.security.pem.PemReader;
+import io.airlift.units.Duration;
+import io.prestosql.plugin.hive.authentication.HiveMetastoreAuthentication;
+import io.prestosql.spi.NodeManager;
+import org.apache.thrift.transport.TTransportException;
+
+import javax.inject.Inject;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+import javax.security.auth.x500.X500Principal;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateExpiredException;
+import java.security.cert.CertificateNotYetValidException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static java.lang.Math.toIntExact;
+import static java.util.Collections.list;
+import static java.util.Objects.requireNonNull;
+
+public class DefaultThriftMetastoreClientFactory
+        implements ThriftMetastoreClientFactory
```

Comment:
Split this commit into rename + add interface, so that git tracks changes correctly.